assets/app/view/route_selector.rb

```diff
@@ -17,9 +17,32 @@ def active_routes
       @routes.select { |r| r.connections.any? }
     end
 
+    def last_run
+      trains = @game.round.current_entity.trains
+      last_run = @game.round.current_entity.last_run
+      return [] unless last_run
+
+      last_run.map do |train, connections|
+        connection_hexes = connections&.map do |ids|
+          ids.map { |id| @game.hex_by_id(id) }
+        end
+        # A future enhancement to this could be to find trains and move the routes over
+        Engine::Route.new(@game.phase, train, connection_hexes: connection_hexes) if trains.include?(train)
+      end.compact
+    end
+
     def render
       trains = @game.round.current_entity.trains
 
+      if @routes.empty?
```

Comment:
this looks awkward, the code will probably change when we move last_revenue to all revenues but if you were to keep this

it should be like

@routes = last_run if @routes.empty

...

Reply:
You'd need to store the routes and selected routes so the map updates correctly.